src/plugins/data/public/query/timefilter/get_time.ts

```diff
@@ -32,6 +32,44 @@ export function calculateBounds(timeRange: TimeRange, options: CalculateBoundsOp
   };
 }
 
+function createTimeRangeFilter(
+  indexPattern: IIndexPattern,
+  timeRange: TimeRange,
+  field: IFieldType,
+  forceNow?: Date
+) {
+  const bounds = calculateBounds(timeRange, { forceNow });
+  if (!bounds) {
+    return;
+  }
+  return buildRangeFilter(
+    field,
+    {
+      ...(bounds.min && { gte: bounds.min.toISOString() }),
+      ...(bounds.max && { lte: bounds.max.toISOString() }),
+      format: 'strict_date_optional_time',
+    },
+    indexPattern
+  );
+}
+
+export function getTimeForField(
+  indexPattern: IIndexPattern | undefined,
+  timeRange: TimeRange,
+  fieldName: string
+) {
+  if (!indexPattern) {
+    return;
+  }
+
+  const field = indexPattern.fields.find(f => f.name === fieldName);
+  if (!field) {
+    return;
+  }
+
+  return createTimeRangeFilter(indexPattern, timeRange, field);
+}
+
 export function getTime(
```

Comment:
++ The naming `getTime` and `getTimeFilter` is a bit unclear.

another nit - It would also be nice to maybe add an explanatory comment to each of the functions here.

Reply:
Just took a harder look at this, thanks for the prodding to do this. The naming confusion was because both functions were basically 90% similar code with one extra input, so I've condensed these into the same function with an `options` argument instead of positional arguments.